### ISSUE TITLE
It is not possible to rename a branch if only the casing of letters changes

### ISSUE THREAD

**Issue by DominikPalo** [ForkIssues/Tracker#2169]
Version 2.44.3

### Steps to replicate:

1. Create a branch named `foo`
2. Right-click on the created branch and select "Rename 'foo'"
3. Enter a new name where only the casing of some letters changes (like from lower-case to upper-case)
4. The "Rename" button stays disabled, so the branch cannot be renamed

(As a workaround I have to rename `foo` to `foo2` and then to `Foo`... or use git commands for renaming)
<img width="536" alt="image" src="https://github.com/user-attachments/assets/b1b3387c-7ae3-44fa-bb30-d73e3a5b7f19">

**Comment by DanPristupov** [comment]
No, it's not possible to do this because of the case-insensitive file system restrictions.

> As a workaround I have to ... use git commands for renaming

It's not possible even in the command line:

```
% git branch b1
% git branch -m b1 B1                                                        
fatal: a branch named 'B1' already exists
```


https://github.com/fork-dev/Tracker/issues/906#issuecomment-585054071
https://github.com/fork-dev/Tracker/issues/670
https://github.com/fork-dev/TrackerWin/issues/2144

**Comment by DominikPalo** [comment]
@DanPristupov thank you and thanks for the explanation, now I understand. And sorry, I tried just the first approach (`foo` → `foo2` → `Foo`, but not `git branch -m foo Foo`).

Anyway, in such a case, I'd consider adding a warning message to the "Rename Local Branch" dialog to prevent future bug reports -- when the entered "New name" differs only in letter casing, show something like:

:warning: _The branch cannot be renamed because of the case-insensitive file system restrictions. Use a different name._

**Comment by DanPristupov** [comment]
> Anyway, in such a case, I'd consider adding a warning message to the "Rename Local Branch" dialog to prevent future bug reports

Well, recognizing if file system is case-insensitive is not as easy as it sounds :). Even on mac there are many of them (APFS, HFS+ can be both case sensitive and not), NTFS, exFAT, FAT32, network drives and macFUSE. I guess there is a reason why git itself doesn't show a more helpful message.

P.S. Since we are here. This limitation is solved in the new reference format in git 2.45 (this format is not supported by Fork yet): https://about.gitlab.com/blog/2024/04/30/whats-new-in-git-2-45-0/#reftables-a-new-backend-for-storing-references
